crates/bevy_animation/src/lib.rs

```diff
@@ -1171,3 +1322,291 @@ impl AnimationEvaluationState {
         Ok(())
     }
 }
+
+/// All the events from an [`AnimationClip`] that occurred this tick.
+#[derive(Debug, Clone)]
+struct TriggeredEvents<'a> {
+    direction: TriggeredEventsDir,
+    lower: &'a [AnimationEventKey],
+    upper: &'a [AnimationEventKey],
+}
+
+impl<'a> TriggeredEvents<'a> {
+    fn new(
```

Comment:
Nit: This could probably have a better name than `new` considering how it's being constructed.

Reply:
renamed to `from_animation`